starlette/middleware/base.py

```diff
@@ -14,6 +14,68 @@
 T = typing.TypeVar("T")
 
 
+class _CachedRequest(Request):
+    """
+    If the user calls Request.body() from their dispatch function
+    we cache the entire request body in memory and pass that to downstream middlewares,
+    but if they call Request.stream() then all we do is send an
+    empty body so that downstream things don't hang forever.
+    """
+
+    def __init__(self, *args: typing.Any, **kwargs: typing.Any) -> None:
```

Comment:
Because I don't want to repeat a bunch of boilerplate. I'm just adding two attributes, not interacting with those other parameters at all. This is a very reasonable use of args kwargs.

Reply:
Isn't it only scope, receive and send? 🤔

Reply:
okay I'll name them

Reply:
done